hpedockerplugin/file_manager.py

```diff
@@ -447,35 +558,39 @@ def remove_share(self, share_name, share):
         cmd = cmd_deleteshare.DeleteShareCmd(self, share)
         return cmd.execute()
 
-    def remove_snapshot(self, share_name, snapname):
-        pass
-
-    def get_share_details(self, share_name, db_share):
-        # db_share = self._etcd.get_vol_byname(share_name,
-        #                                      name_key1='shareName',
-        #                                      name_key2='shareName')
-        # LOG.info("Share details: %s", db_share)
-        # if db_share is None:
-        #     msg = (_LE('Share Get: Share name not found %s'), share_name)
-        #     LOG.warning(msg)
-        #     response = json.dumps({u"Err": ""})
-        #     return response
-
-        err = ''
-        mountdir = ''
-        devicename = ''
+    @staticmethod
+    def _rm_implementation_details(db_share):
+        LOG.info("Removing implementation details from share %s..."
+                 % db_share['name'])
+        share = copy.deepcopy(db_share)
+        share.pop("nfsOptions")
+        share.pop("quota_id")
+        share.pop("id")
+        share.pop("readonly")
+        share.pop("comment")
+        LOG.info("Implementation details removed: %s" % share)
+        return share
 
+    @staticmethod
+    def get_share_details(share_name, db_share):
+        # TODO: mount_dir to be fixed later
         path_info = db_share.get('share_path_info')
```

Comment:
Do we have 'share_path_info' key if not this needs to be corrected?

Reply:
We don't... This will be addressed as part of #609

Reply:
This has been addressed in this PR only.